### DISCUSSION TITLE
Problem Shoot 大佬请帮忙看下问题

### DISCUSSION THREAD

**Discussion by chenericliu** [wenda-LLM/wenda#401] [Q&A]
各位大佬，
M40 24GB的卡，已经装了CUDA，并安装了Visual Studio。安装闻达前，已经装了Anaconda。
下载6月21日主程序和5月16日版安装依赖后，一直运行不成功。
得到以下提示：
请各位大佬给看看，拜谢！

------------------------------------------------------------------------------------------
检测到集成环境
Namespace(Config='config.yml', Port=None, Logging=None, LLM_Type='glm6b')
library:
  count: 5
  step: 2
  strategy: calc:2 rtst:2 agents:0
librarys:
  bing:
    count: 5
  bingsite:
    count: 5
    site: www.12371.cn
  fess:
    count: 1
    fess_host: 127.0.0.1:8080
  kg:
    count: 5
    graph_host: ''
    knowledge_path: ''
    model_path: ''
  qdrant:
    batch_size: 32
    collection: qa_collection
    count: 3
    device: cuda
    model_path: model/m3e-base
    overlap: 0
    path: txt
    qdrant_host: http://localhost:6333
    similarity_threshold: 0.8
    size: 20
  remote:
    host: http://127.0.0.1:17860/api/find
  rtst:
    count: 3
    device: cuda
    model_path: model/m3e-base
    overlap: 0
    size: 20
llm:
  path: model/chatglm-6b-int4
  strategy: cuda fp16
llm_type: glm6b
logging: false
port: 17860

[['cuda', 'fp16']]
No sentence-transformers model found with name model/m3e-base. Creating a new one with MEAN pooling.
知识库加载完成
No compiled kernel found.
Compiling kernels : C:\Users\Administrator\.cache\huggingface\modules\transformers_modules\local\quantization_kernels_parallel.c
Compiling gcc -O3 -fPIC -pthread -fopenmp -std=c99 C:\Users\Administrator\.cache\huggingface\modules\transformers_modules\local\quantization_kernels_parallel.c -shared -o C:\Users\Administrator\.cache\huggingface\modules\transformers_modules\local\quantization_kernels_parallel.so
'gcc' is not recognized as an internal or external command,
operable program or batch file.
Compile default cpu kernel failed, using default cpu kernel code.
Compiling gcc -O3 -fPIC -std=c99 C:\Users\Administrator\.cache\huggingface\modules\transformers_modules\local\quantization_kernels.c -shared -o C:\Users\Administrator\.cache\huggingface\modules\transformers_modules\local\quantization_kernels.so
'gcc' is not recognized as an internal or external command,
operable program or batch file.
Compile default cpu kernel failed.
Failed to load kernel.
Cannot load cpu kernel, don't use quantized model on cpu.
Using quantization cache
Applying quantization to glm layers
模型加载完成
-----------------------------------------------------------------------------------------------------------------

**Comment by l15y** [comment]
没看到有问题

**Comment by chenericliu** [comment]
作者您好，


可能是我的帖子没有描述清楚，给您截了图，


错误信息为：错误 CUDA Error: no kernel image is available for execution on the device


看起来像是一个CUDA的问题，我尝试重装了CUDA，并且重装了torch，依然没有解决。


是我的m40卡不支持cuda吗？







------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "wenda-LLM/wenda"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2023年6月23日(星期五) 晚上8:19
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [wenda-LLM/wenda] Problem Shoot 大佬请帮忙看下问题 (Discussion #401)





 
没看到有问题

**Reply by l15y** [reply]
说明设备与cuda版本不兼容。重装cuda后还需要更换与cuda对应的torch版本
